docs: Document ROFL proxy custom domains (#2349)
oasisprotocol/oasis-sdk: docs/rofl/features/proxy.mdx

      - "5678:5678" # Expose container port 5678 on host port 5678
```

> After deploying your ROFL app, you can find the generated URL by

Reviewer: ```suggestion
After deploying your ROFL, you can find the generated URL by
```

The proxy behavior can be configured using annotations in
your `compose.yaml` file.

### Overview

Each annotation follows this general format:

`net.oasis.proxy.ports.<published_port>.<setting>: <value>`

Where:

`<published_port>` — the external port exposed in your compose.yaml

`<setting>` — the specific proxy configuration (e.g., mode, custom_domain)

### Example Configuration

> The following example configures a service to use TCP passthrough with a custom domain:

Reviewer 2: ```suggestion
The following example configures port 8080 to use TCP passthrough,
and port 80 to use terminate-TLS mode (the default) with a custom domain.
```

```yaml
title="compose.yaml"
services:
  myservice:
    image: docker.io/my/service:latest
    ports:
      - "8080:8080"
      - "80:80"
    annotations:
      net.oasis.proxy.ports.8080.mode: passthrough
      net.oasis.proxy.ports.80.custom_domain: mydomain.com
```

Explanation:

> - The container exposes ports 8080 and 80.

Reviewer 2: ```suggestion
- ROFL App exposes ports 8080 and 80.
```

- The proxy forwards the raw TCP connection to your container (mode: passthrough).

> - Requests to mydomain.com are routed to this service (custom_domain: mydomain.com).

Reviewer 2: This is maybe a bit to simlpified. See my other comment about `custom_domain` setting.
